airflow/ti_deps/deps/ready_to_reschedule.py

```diff
@@ -44,7 +45,8 @@ def _get_dep_statuses(self, ti, session, dep_context):
         from airflow.models.mappedoperator import MappedOperator
 
         is_mapped = isinstance(ti.task, MappedOperator)
-        if not is_mapped and not getattr(ti.task, "reschedule", False):
+        is_debug_executor = conf.get("core", "executor") == "DebugExecutor"
```

Comment:
I wonder if there’s a better way to obtain this information. But since this pattern is used elsewhere inthe code base, it’s probably good enough.

Reply:
@o-nikolas - maybe we could also add "is_debug" as part of AIP-51 ?

Reply:
Yeah not great to pull this from a global static place, but looking through what we have accessible here, I don't see anything that could help us. And injecting this all the way through here looks like a lot of pain.

I'm pushing a commit to use the constant that's available, making this piece of code a tiny bit better I think ?

Reply:
> @potiuk this is covered by the single threaded case actually, which is more precisely what the issue is. Any single threaded executor needs to reschedule sensors otherwise the thread will be blocked when the sensor sleeps.

Yep. You are right. Single-threaded is the right check here and we should use it here. I think being single-threaded is the only reason we are doing it.

Reply:
ok, I just checked, and the SequentialExecutor has the same Behavior, in that it does not wait to reschedule, it's just slower at it so it spams less requests.
As such, it appears that checking on single threaded is indeed the way to go.

Reply:
I'm a bit confused by `DebugExecutor`. Didn't we deprecate it with the attention of removing it completely in Airflow 3?
https://github.com/apache/airflow/issues/28861#issuecomment-1378905294

Reply:
Yeah it was removed in the context of the `dag.test()` command in [this commit](https://github.com/apache/airflow/commit/71c64de96248694017897fdb3d9d241e7c980825). But AIP-47 compliant system tests still depend on that executor, so we can't kill it just yet without finding a replacement for that.

Reply:
What`s mor i think we do not possibly even want to replace it for AIP-47. Why would we? Maybe we can rename to SystemTestExcecutor but I think it does the job nicely ?

Reply:
A flag for executors that require reschedule mode has been added in #28934, I've just merged it, you can now make use of it here @vandonr-amz